client/src/app/components/ToolbarBulkSelector.tsx

```diff
@@ -1,152 +1,139 @@
-import React, { useState } from "react";
+import React, { useState, useMemo } from "react";
 import { useTranslation } from "react-i18next";
 import {
-  Button,
   Dropdown,
   DropdownItem,
   DropdownList,
   MenuToggle,
   MenuToggleCheckbox,
-  PaginationProps,
   ToolbarItem,
 } from "@patternfly/react-core";
 
-import AngleDownIcon from "@patternfly/react-icons/dist/esm/icons/angle-down-icon";
-import AngleRightIcon from "@patternfly/react-icons/dist/esm/icons/angle-right-icon";
-
-export interface IToolbarBulkSelectorProps<T> {
+export interface IToolbarBulkSelectorProps {
   areAllSelected: boolean;
-  areAllExpanded?: boolean;
-  onSelectAll: (flag: boolean) => void;
-  onExpandAll?: (flag: boolean) => void;
-  selectedRows: T[];
-  onSelectMultiple: (items: T[], isSelecting: boolean) => void;
-  currentPageItems: T[];
-  paginationProps: PaginationProps;
-  isExpandable?: boolean;
+  itemCounts: {
+    items: number;
+    filtered?: number;
+    page: number;
+    selected: number;
+  };
+  onSelectNone: () => void;
+  onSelectCurrentPage: () => void;
+  onSelectAllFiltered?: () => void;
+  onSelectAll?: () => void;
 }
 
-export const ToolbarBulkSelector = <T,>({
-  currentPageItems,
+/**
+ * Selection options:
+ *   - Default Button / Select Page
+ *   - Split / None, Count: 0
+ *   - Split / Page (every record on the current page), Count: records in current page
+ *   - Split / All Filtered (every record matching the filter), Count: filtered records
+ *   - Split / All (every record, every page), Count: items size
```

Comment:
we might not be able to deliver this feature in every case i.e. on screens with server-side paging or with incomplete/stale data. We should be able to toggle the feature for screens that we know that will work.

Reply:
Yes, the "all filtered" and the "all" actions are optional and are only included if the handler is provided.